### ISSUE TITLE
Instructions how to install pre-releases

### ISSUE THREAD

**Issue by dbeatty10** [fishtown-analytics/docs.getdbt.com#2144]
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Updates could possible go on this page: https://docs.getdbt.com/dbt-cli/install/pip

I would love to see two things:
1. [Instructions to create a python virtual environment #2143](https://github.com/dbt-labs/docs.getdbt.com/discussions/2143)
    - Would be nice is this was a stand-alone page
3. Instructions to install pre-releases of dbt adapters
    - Could be stand-alone or an independent section within https://docs.getdbt.com/dbt-cli/install/pip

To install pre-releases for bigquery, databricks, and snowflake all at the same time within `zsh`, it would look like this:

```shell
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
```

But of course we actually want to do this within a virtual environment, like so 😉 
```shell
dbt --version
python3 -m venv venv
source venv/bin/activate
python3 -m pip install --upgrade pip
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
source venv/bin/activate
dbt --version
```

Ideally, we'd provide different instructions based on the users's operating system and shell, like this:

<details open>
<summary>POSIX bash/zsh</summary>

```shell
dbt --version
python3 -m venv venv
source venv/bin/activate
python3 -m pip install --upgrade pip
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
source venv/bin/activate
dbt --version
```
</details>

<details>
<summary>POSIX fish</summary>

```shell
python3 -m venv venv
source venv/bin/activate.fish
python3 -m pip install --upgrade pip
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
source venv/bin/activate.fish
```
</details>

<details>
<summary>POSIX csh/tcsh</summary>

```shell
python3 -m venv venv
source venv/bin/activate.csh
python3 -m pip install --upgrade pip
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
source venv/bin/activate.csh
```
</details>

<details>
<summary>POSIX PowerShell Core</summary>

```shell
python3 -m venv venv
venv/bin/Activate.ps1
python3 -m pip install --upgrade pip
python3 -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
venv/bin/Activate.ps1
```
</details>

<details>
<summary>Windows cmd.exe</summary>

```shell
python -m venv venv
venv\Scripts\activate.bat
python -m pip install --upgrade pip
python -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
venv\Scripts\activate.bat
```
</details>

<details>
<summary>Windows PowerShell</summary>

```shell
python -m venv venv
venv\Scripts\Activate.ps1
python -m pip install --upgrade pip
python -m pip install --pre dbt-core dbt-bigquery dbt-databricks dbt-snowflake
venv\Scripts\Activate.ps1
```
</details>

### Use cases
This gives users clear instructions that they can copy-paste to use beta releases or release candidates.

### Expected outcome
User has dbt installed in an isolated environment that doesn't affect the rest of their system. Easy to deactivate and not use if it contains bugs.

### Example before vs. after

Before:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/44704949/194158247-a6f672ae-d495-4725-9f54-319e5192e4c4.png">

After:
<img width="433" alt="image" src="https://user-images.githubusercontent.com/44704949/194158478-84cb2bf2-31e4-4c3b-95bd-37778657ebda.png">

**Comment by runleonarun** [comment]
@dbeatty10 I am closing this because the install has changed and this is really old. Let me know if you prefer either:
* Add the icebox label to deal with it and some later time when our queue is smaller, or
* Leave it open in hopes it can get addressed

**Comment by dbeatty10** [comment]
Thanks for following-up on this @runleonarun !

I think it would still be helpful to provide instructions how to install pre-releases. Maybe we want to do that through [`dbt-core-bundles`](https://github.com/dbt-labs/dbt-core-bundles?tab=readme-ov-file#how-to-use)?

How would you feel about re-opening this with the `icebox` label?

**Comment by mirnawong1** [comment]
thanks @dbeatty10 , happy to take this soon but wanted to double check if this was still the case given the recent 1.8 core + adapter installation update. it looks like this matches (e.g. `pip install dbt-core dbt-snowflake`) though but triple checking!

**Comment by dbeatty10** [comment]
Yep, I think these commands will work for v1.8+ 👍 But let me know if not and I will be happy to help troubleshoot.

Note: for folks upgrading from <= v1.7 to >=v1.8, they may need to do something like [this](https://github.com/dbt-labs/dbt-core/issues/10135#issuecomment-2113728550) also.

**Comment by nataliefiann** [comment]
Hiya @dbeatty10 

I just wanted to double check with the workaround mentioned on your [ticket](https://github.com/dbt-labs/dbt-core/issues/10135#issuecomment-2113728550), will users on 1.7 need to upgrade to 1.8 for a pre-release or do they need to use the workaround you mentioned in your ticket?

Kind Regards
Natalie

**Comment by dbeatty10** [comment]
> I just wanted to double check with the workaround mentioned on your [ticket](https://github.com/dbt-labs/dbt-core/issues/10135#issuecomment-2113728550), will users on 1.7 need to upgrade to 1.8 for a pre-release or do they need to use the workaround you mentioned in your ticket?

The workaround mentioned [here](https://github.com/dbt-labs/dbt-core/issues/10135#issuecomment-2113728550) may be necessary when upgrading directly from a version less than v1.8 to v1.8 or above.

If installing a pre-release into a fresh virtual environment that doesn't already have dbt installed, it won't be needed.
